### ISSUE TITLE
Build bc-bytewords with bc-crypto-base without install

### ISSUE THREAD

**Issue by ChristopherA** [BlockchainCommons/bc-bytewords#9]
Low priority, but can we automate the build of bc-bytewords so that we don't have to separately `sudo make install` bc-crypto-base?

**Comment by wolfmcnally** [comment]
I've added this to my backlog.

**Comment by wolfmcnally** [comment]
After consideration, the main issue with building libraries (as opposed to command line tools) is that to be useful, libraries *need* to be installed on the system. So in the case of building bc-bytewords, it depends on bc-crypto-base, the same as it depends on the standard C library for some of its functions. But building bc-bytewords doesn't link or combine it with bc-crypto-base— once it's installed it's just another library sitting there on the system next to bc-crypto-base waiting to be linked with by other build processes. And just like the standard C library, the bc-crypto-base that bc-bytewords links to needs to pre-exist on the system. All the actual command line tools (bytewords, seedtool, keytool) use Git submodules to bring in, compile, and statically link to the dependent libraries they need, but this approach doesn't really work with raw libraries.

Take, for example, a scenario where we actually did create a build script that pulled in bc-crypto-base as a submodule and then built bc-bytewords. You then `sudo make install` that version of bc-bytewords, but there either 1) might not be a version of bc-crypto-base installed on the system at all or 2) the version that *is* installed is too old or otherwise incompatible with the version of bc-bytewords you just built. There is no good reason to expect that the version of bc-bytewords you just installed would be usable with anything else on the system that wanted to link with it and its dependency, bc-crypto-base.

If our hypothetical build-script built and installed *all* of a library's dependencies (in this example bc-crypto-base) then it would potentially either duplicate the effort to build a version of the library already installed on the system, or could potentially replace the library with an incompatible version, breaking other dependencies.

Basically you can't test build a library without having its dependencies installed. You *can* build a standalone executable like seedtool that can build and statically link to its own copy of the dependent library.
